src/cli/sourcemap.rs

```diff
@@ -178,20 +196,32 @@ fn recurse_create_node<'a>(
         return None;
     }
 
-    let file_paths = instance
+    let file_paths: Vec<&Path> = instance
         .metadata()
         .relevant_paths
         .iter()
         // Not all paths listed as relevant are guaranteed to exist.
         .filter(|path| path.is_file())
         .map(|path| path.strip_prefix(project_dir).expect(PATH_STRIP_FAILED_ERR))
-        .map(|path| path.to_path_buf())
         .collect();
 
+    let mut file_path_buffs: Vec<PathBuf> = Vec::new();
```

Comment:
We can use `Vec<Cow<Path>>` here over `Vec<PathBuf>` to save allocations when the flag isn't passed. It'd involve changing the type of `SourcemapNode` a bit but it isn't a public interface so there's no concern there.